packages/types/src/base.ts

```diff
@@ -57,7 +57,7 @@ export type Asset = {
   explorer: string
   explorerTxLink: string
   explorerAddressLink: string
-  relatedAssetKey?: AssetId
+  relatedAssetKey: AssetId | null | undefined
```

Comment:
The crux of this PR - JSON doesn't support `undefined`.

By making the `relatedAssetKey` required and having it as maybe null or undefined, we ensure that it is either `undefined` (e.g base assets with known no related assets, or generated assets for which we don't know related assets yet, to be fetched by the `generateRelatedAssetIndex` script), or an explicit `null` in the JSON (i.e related asset indexes have been fetched, but there are none).

In other words, by having `AssetId | null` in the JSON, we are now able to not refetch existing related AssetIds with a simple `  if (existingRelatedAssetKey !== undefined) return` check